File: src/modules/MediaControlsModule.ts
import { NativeModule, requireNativeModule } from 'expo-modules-core';
import { Platform } from 'react-native';

export type MediaControlsModuleEvents = Record<string, any>;

declare class MediaControlsModule extends NativeModule<MediaControlsModuleEvents> {
  /**
   * Get whether the media controls integration is enabled for the specified player.
   */
  isEnabled(playerId: string): Promise<boolean | null>;

  /**
   * Enable or disable the media controls integration for the specified player.
   */
  setEnabled(playerId: string, enabled: boolean): Promise<void>;
}

export default Platform.OS === 'ios'
  ? requireNativeModule<MediaControlsModule>('MediaControlsModule')
  : null;

File: android/src/main/java/com/bitmovin/player/reactnative/MediaControlsModule.kt
package com.bitmovin.player.reactnative

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class MediaControlsModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("MediaControlsModule")

        AsyncFunction("isEnabled") { _: String ->
            // The media controls integration is iOS/tvOS-only, always disabled on Android.
            false
        }

        AsyncFunction("setEnabled") { _: String, _: Boolean ->
            // The media controls integration is iOS/tvOS-only, no-op on Android.
        }
    }
}
